### DISCUSSION TITLE
Change dsld interfaces?

### DISCUSSION THREAD

**Discussion by Koteuo** [Freetz-NG/freetz-ng#1036] [Allgemein]
Hi,
I have a question.

I'm running a FRITZ!Box 5490 with Freetz-ng.
I've configured internet access using port 1 (ATA mode), so the fiber port is not being used.
What I want to do is use the fiber port just as a normal LAN port.
Physically everything is connected correctly. On the other end of the fiber cable I have a media converter with an SFP module.
When I add `wan` interface to the `lan` bridge, so that:

> brinterfaces {
>         name = "lan";
>         dhcp = no;
>         ipaddr = 10.0.0.1;                                                                                
>         netmask = 255.0.0.0;                                                                              
>         dstipaddr = 0.0.0.0;                                                                              
>         interfaces = "eth0", "eth1", "eth2", "eth3", "wlan", "plc", "wan";                                
>         dhcpenabled = yes;                                                                                
>         dhcpstart = 10.0.1.100;                                                                           
>         dhcpend = 10.0.1.250;                                                                             
>         is_guest = no;                                                                                    
>         is_hotspot = no;                                                                                  
>         multicast_snooping = yes;                                                                         
>         is_public = no;
> }

The dhcp server starts serving LAN addresses on the fiber port, and I am able to ping LAN devices and internet! Even resolve domains using nslookup!
But I can't do anything else. All other packets are dropped. I guess it's the dsld 's fault.
Is there a way to configure it?
To permit traffic from the 'wan' interface?
or should I compile a new firmware with iptables instead?

**Comment by fda77** [comment]
First its a good idea to test is to disable avm "packet accelerator", see FAQ. Maybe the HW acceleration without linux(-kernel) prevents it.
Then you could take a look into avm kernel sources for your device http://osp.avm.de (freetz has for this device no "replace kernel"!)
Older devices had a pin configuration in the kernel for the switch, which could also be modified with https://freetz-ng.github.io/freetz-ng/make/cpmaccfg.html

**Reply by Koteuo** [reply]
I disabled the avm "packet accelerator" and the packets are coming through!
But the max speed I get is ~60mb/s

**Comment by fda77** [comment]
You could try to reconfigure the avmpa, but it has no manual
try
/proc/avm/hw_pa/
or
/proc/net/avm_pa/
and read the avm sources (unpacked by freetz):
```
find source/kernel/ref-vr9-5491_07.27/  -name *avm* -a ! -type d |grep pa
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/include/avm/pa/avm_pa.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/include/avm/pa/avm_pa_hw.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/include/avm/pa/avm_pa_ifx_multiplexer.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa_selector.c
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa_hw.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa_ifx_multiplexer.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa_intern.h
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa.c
source/kernel/ref-vr9-5491_07.27/linux-3.10/net/avm_pa/avm_pa_sessiongroups.c
source/kernel/ref-vr9-5491_07.27/linux-3.10/drivers/net/avm_cpmac/switch/ifx/common/avm_pa_ifx_multiplexer.c
source/kernel/ref-vr9-5491_07.27/linux-3.10/drivers/char/avm_new/avm_page_statistic.c
```

**Reply by fda77** [reply]
seems instead "disabled" you could use only "fw_disabled"...

**Reply by Koteuo** [reply]
as in `echo fw_disabled > /proc/net/avm_pa/control` ?
Doesn't work. The packet acceleration works, but now the packets aren't coming through.

**Reply by fda77** [reply]
Yes, i found it in kernel sources. If not maybe somwhere else ... dont know

**Reply by Koteuo** [reply]
Im looking through the avm_pa sources and there are comments explaining a lot of stuff.
for example:
I may have missed it, if it was mentioned somewhere in the FAQs, but I never knew there was a way to filter accelerated sessions.

> * For example a user can do the following:
>  * echo ip > /proc/net/avm_pa/xsession
>  * echo port 80 > /proc/net/avm_pa/xsession
>  * echo ip6 dst port 25  > /proc/net/avm_pa/xsession
>  * echo not vpid internet > /proc/net/avm_pa/xsession
>  *
>  * /proc/net/avm_pa/sessions will show only sessions matching the criterea. Other sessions
>  * are hidden, but still exist (and packets of hidden sessions are still accelerated).

**Reply by fda77** [reply]
in the faq is only to disable via proc, nobody yet read the whole sources and extracted usefull things...
eg like i did it for ctlmgt_ctl https://freetz-ng.github.io/freetz-ng/wiki/60_Development/uimods.html

**Reply by Koteuo** [reply]
I found a piece of code responsible for the `/proc/net/avm_pa/brief` file output generation.

```
> static void pa_show_brief_status_header(pa_fprintf fprintffunc, void *arg)
> {
>    struct avm_pa_global *ctx = &pa_glob;
>    const char *mode;
> 
>    if (ctx->disabled)
>       mode = "disabled";
>    else if (ctx->fw_disabled)
>       mode = "testmode";
>    else if (avm_pa_capture_running())
>       mode = "capture";
>    else
>       mode = "enabled";
>    (*fprintffunc)(arg, "State          : %s\n", mode);
> 
>    if (ctx->hardware_pa.add_session) {
>       mode = ctx->hw_ppa_disabled ? "disabled" : "enable";
>       (*fprintffunc)(arg, "HW State       : %s\n", mode);
>    }
> 
>    if (ctx->filter_enabled && list_empty(&ctx->accel_filter))
>       mode = "empty";
>    else if (ctx->filter_enabled)
>       mode = "yes";
>    else
>       mode = "no";
>    (*fprintffunc)(arg, "Filter active  : %s\n", mode);
> }
```

When the `avm_pa` is disabled it's supposed to output `disabled`, the same goes for when it's enabled, but when it is in `fw_disabled` mode it's supposed to output `testmode`.

But when I `fw_disabled > /proc/net/control`, it doesn't change state, regardless if I disable it or enable if first!

```
> root@fritz:/proc/net/avm_pa# cat brief
> Version        : For Linux version 3.10.107 (jenkins@jenkins-ci.avm.de) (collect2: error: ld returned 1 exit status) #1 SMP 2021-05-26
> **State          : disabled**
> HW State       : enable
> BSession allow : yes
> Filter active  : empty
> BSessions      : 126
> Sessions       : 0
> HW Sessions    : 0
> Max Sessions   : 689
> Max Bulk HW Ses: 2048
> Sessions (dead): 0
> Sessions (free): 2047
> Egress pool    : 40/40
> Queuelen       : 0
> Rx pkts/secs   : 57
> Fw pkts/sec    : 13
> Ov pkts/sec    : 0
> Rx pakets      : 2278284
> Rx bypass      : 254360
> Rx ttl <= 1    : 0
> Rx broadcast   : 0
> Rx search      : 2002195
> Rx match       : 1263372
> Rx modified    : 1028555
> Fw pakets      : 1220918
> Fw local       : 48342
> VPID1 : RX      75274 TX      56722 wan
> VPID2 : RX          0 TX          8 eth0
> VPID3 : RX    1286422 TX     520087 eth1
> VPID4 : RX          0 TX          0 eth2.2
> VPID5 : RX          0 TX          0 eth3.3
> VPID6 : RX     151603 TX     394448 wasp
> VPID7 : RX    1083834 TX     807354 vlan_master0
> VPID8 : RX     141804 TX     813419 internet
```

What also seems strange is, that when I enable the `avm_pa` and I don't get any connection, it doesn't change the "discard" or "error" statistics.
<img width="780" alt="Zrzut ekranu 2024-09-8 o 15 19 00" src="https://github.com/user-attachments/assets/766613c0-77dc-4e20-983a-1a3dda395071">

Maybe it accepts the traffic, but it doesn't get routed correctly by it?
Or maybe it doesn't see it at all?
I'll see what else I can find

**Comment by fda77** [comment]
Always keep in mind that avm "gpl sources" could no be the sources avm really used to build (who cares about gpl?)
Just search for "crappy avm" commit comments: https://github.com/search?q=repo%3AFreetz-NG%2Ffreetz-ng+crappy+avm&type=commits In all these cases (and even many more with other commit commens) the kernel .config does not match the kernel sources, fromtghe same pack avm released! The are missing or new symbols... nobody knows with wich setting avm build the kernel! So the best for AVMPA is the source but better than nothing

This pice of code is also responsible avm can not do normal linux things, eg VLAN support does never come or many changes of AVMPA are neede which then will again cause many bugs .. as always

Also because of AVMPA you device could only be PINGed on WAN ip only be 1 IP at the same time! eg if you ping your wan with 3 different ips, 2 dont get an answer! (no always the same device). This problem exists since a decade and avms "fix" is the "stealth mode" which does not answer any more any pings ^^

And because of AVMPA i do not use anymore avm software on my router (compatible avm hardware runs great with openwrt...)
